src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationAttribute.cs

```diff
@@ -252,54 +249,48 @@ private void SetupResourceAccessor()
             {
-                var property = _errorMessageResourceType
-                    .GetTypeInfo().GetDeclaredProperty(_errorMessageResourceName);
-                if (property != null && !ValidationAttributeStore.IsStatic(property))
-                {
-                    property = null;
-                }
-
-                if (property != null)
-                {
-                    var propertyGetter = property.GetMethod;
-
-                    // We only support internal and public properties
-                    if (propertyGetter == null || (!propertyGetter.IsAssembly && !propertyGetter.IsPublic))
-                    {
-                        // Set the property to null so the exception is thrown as if the property wasn't found
-                        property = null;
-                    }
-                }
+                property = null;
+            }
 
-                if (property == null)
-                {
-                    throw new InvalidOperationException(
-                        string.Format(
-                            CultureInfo.CurrentCulture,
-                            SR.ValidationAttribute_ResourceTypeDoesNotHaveProperty,
-                            _errorMessageResourceType.FullName,
-                            _errorMessageResourceName));
-                }
+            if (property != null)
+            {
+                var propertyGetter = property.GetMethod;
 
-                if (property.PropertyType != typeof(string))
+                // We only support internal and public properties
+                if (propertyGetter == null || (!propertyGetter.IsAssembly && !propertyGetter.IsPublic))
                 {
-                    throw new InvalidOperationException(
-                        string.Format(
-                            CultureInfo.CurrentCulture,
-                            SR.ValidationAttribute_ResourcePropertyNotStringType,
-                            property.Name,
-                            _errorMessageResourceType.FullName));
+                    // Set the property to null so the exception is thrown as if the property wasn't found
+                    property = null;
                 }
+            }
 
-                _errorMessageResourceAccessor = delegate { return (string)property.GetValue(null, null); };
+            if (property == null)
+            {
+                throw new InvalidOperationException(
+                    string.Format(
+                        CultureInfo.CurrentCulture,
+                        SR.ValidationAttribute_ResourceTypeDoesNotHaveProperty,
+                        _errorMessageResourceType.FullName,
+                        _errorMessageResourceName));
             }
-            else
+
+            if (property.PropertyType != typeof(string))
             {
-                throw new InvalidOperationException(string.Format(CultureInfo.CurrentCulture,
```

Comment:
This method is called from the method SetupResourceAccessor(), which does the following:
```
private void SetupResourceAccessor()
{
	bool resourceNameSet = !string.IsNullOrEmpty(_errorMessageResourceName);
	bool resourceTypeSet = _errorMessageResourceType != null;
	// ...
	if (resourceNameSet != resourceTypeSet) throw new InvalidOperationException();
	if (resourceNameSet)
	{
	    SetResourceAccessorByPropertyLookup();
	}
    else 
    {
        // ...
    }
}
```
This is the only point that `SetResourceAccessorByPropertyLookup` is called.
If resourceNameSet is true, then resourceTypeSet has to be true when we call the method, as we throw an InvalidOperationException if this is not the case.
Thus, when we call `SetResourceAccessorByPropertyLookup`, and check `if (_errorMessageResourceType != null && !string.IsNullOrEmpty(_errorMessageResourceName))`, this case will always be true, as:
```
resourceTypeSet == true, therefore _errorMessageResourceType != null
resourceNameSet == true, therefore !string.IsNullOrEmpty(_errorMessageResourceName)
```